.github/workflows/ci-test.yml

```diff
@@ -56,6 +56,34 @@ jobs:
         run: |
           ./contribution/k3s/install_k3s.sh
 
+      - name: Setup self-hosted docker registry
+        run: |
```

Comment:
Can we move these to scripts instead? We would be needing them local dev environment too right?

Reply:
Yes, its a good idea. I have implimented the required changes. Please take an other look.